lib/ansible/plugins/filter/core.py

```diff
@@ -229,7 +229,7 @@ def from_yaml_all(data):
 
 
 @pass_environment
-def rand(environment, end, start=None, step=None, seed=None):
+def rand(environment, end, weights=None, cum_weights=None, k=1, start=None, step=None, seed=None):
```

Comment:
This needs a [changelog](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and test.

new options should be added to the end of the function
```suggestion
def rand(environment, end, start=None, step=None, seed=None, weights=None, cum_weights=None, k=1):
```